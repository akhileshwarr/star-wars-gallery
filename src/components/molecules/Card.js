import React,  { useContext, useCallback } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CardMet from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import { Text } from '../atoms/Text';
import { ThemeContext } from '../../context/theme';
import { SWButton as Button } from '../atoms/SWButton'
import { useDispatch } from "react-redux";
import { fetchFilim } from '../../store/actions'
import PropTypes from 'prop-types';


const useStyles = makeStyles({
  root: {
    minWidth: 275,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

export const  Card = ({person}) => {

  const theme = useContext(ThemeContext);
  const classes = useStyles();
  const dispatch = useDispatch()
  const handleFilmClick = useCallback(
      () => dispatch(fetchFilim(person.url)),
      [dispatch]
    )
  return (
    <CardMet className={classes.root} style = {{backgroundColor: theme.backgroundCard }}>
      <CardContent>
        <Typography  color="textSecondary"  variant="h5" component="h2">
            <Text>{person.name}</Text>
        </Typography>
        <Typography className={classes.title} >
          <Text>Height : {person.height}</Text> 
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
            <Text> Gender : {person.gender}</Text>
        </Typography>
        <Typography variant="body2" component="p">
            <Text>Films:</Text> <br />
            {
              person.filimDetails ? 
                person.filimDetails.map((film,i) =>  
                  <React.Fragment key = {i}>
                    <Text>{`Film${i} : `}</Text>
                    <br/>
                    <Text>{"Title : "}{film.title}</Text>
                    <br/>
                    <Text>{"Director : "}{film.director}</Text>
                    <br/>
                    <Text>{"producer : "}{film.director}</Text>
                    <br/>
                  </React.Fragment>
                )
                :person.films.map((a,i) => <React.Fragment key = {i}>
                    <Text>{a}</Text>
                    <br/>
                    </React.Fragment>)
                }
        </Typography>
      </CardContent>
      <CardActions>
        <Button handleSWClick = {handleFilmClick} disabled = { person.filimDetails ? true : false } > 
          <Text>Learn More</Text>
        </Button>
      </CardActions>
    </CardMet>
  );
}

Card.propTypes = {
  person: PropTypes.object
};

Card.defaultProps = {
  person: {},
 
};
