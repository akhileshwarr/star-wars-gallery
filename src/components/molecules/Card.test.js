import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from './Card.js';
import { Provider } from 'react-redux'
import rootReducer from '../../store/reducers'
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux'

const store = createStore(rootReducer, applyMiddleware(thunk))

test('/Card renders snapshot', () => {
    const tree = renderer.create(<Provider store={store}>
           <Card person = {{
                name: "sample",
                height: "sample",
                gender: "gender",
                films:["filim1"],
                url:"asdf"
              }} />
   </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
