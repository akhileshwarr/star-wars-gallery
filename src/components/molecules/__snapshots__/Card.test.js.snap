// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Card renders snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
    >
      <span
        className="genralSpan"
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        sample
      </span>
    </h2>
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
    >
      <span
        className="genralSpan"
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        Height : 
        sample
      </span>
    </p>
    <p
      className="MuiTypography-root makeStyles-pos-4 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <span
        className="genralSpan"
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
         Gender : 
        gender
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      <span
        className="genralSpan"
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        Films:
      </span>
       
      <br />
      <span
        className="genralSpan"
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        filim1
      </span>
      <br />
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="genralSpan"
          style={
            Object {
              "color": "#ffffff",
            }
          }
        >
          Learn More
        </span>
      </span>
    </button>
  </div>
</div>
`;
